fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/adapters/EventExpandableListViewAdapter.java

```diff
@@ -264,10 +306,12 @@ public View getChildView(
    */
   private void putEventsInMap(List<Event> events) {
     Collections.sort(events);
-
+    this.eventsMap.get(PAST).clear();
```

Comment:
I don't understand why you clear the previous events in PAST,PRESENT,FUTURE ? Didn't it worked without doing that ?